slither/tools/kspec_coverage/analysis.py

```diff
@@ -0,0 +1,215 @@
+from slither.solc_parsing.declarations.function import Function
+from slither.core.declarations.solidity_variables import SolidityFunction
+from tabulate import tabulate
+
+def get_all_covered_kspec_functions(target):
+    # Create a set of our discovered functions which are covered
+    covered_functions = set()
+
+    BEHAVIOUR_PATTERN = re.compile('behaviour\s+(\S+)\s+of\s+(\S+)')
+    INTERFACE_PATTERN = re.compile('interface\s+([^\r\n]+)')
+
+    def refactor_type(type):
+        return {
+            'uint': 'uint256',
+            'int': 'int256'
+        }.get(type, type)
+
+    # Read the file contents
+    with open(target, 'r', encoding='utf8') as target_file:
+        lines = target_file.readlines()
+
+    # Loop for each line, if a line matches our behaviour regex, and the next one matches our interface regex,
+    # we add our finding
+    i = 0
+    while i < len(lines):
+        match = BEHAVIOUR_PATTERN.match(lines[i])
+        if match:
+            contract_name = match.groups()[1]
+            match = INTERFACE_PATTERN.match(lines[i + 1])
+            if match:
+                function_full_name = match.groups()[0]
+                start, end = function_full_name.index('(') + 1, function_full_name.index(')')
+                function_arguments = function_full_name[start:end].split(',')
+                function_arguments = [refactor_type(arg.strip().split(' ')[0]) for arg in function_arguments]
+                function_full_name = function_full_name[:start] + ','.join(function_arguments) + ')'
+                covered_functions.add((contract_name, function_full_name))
+                i += 1
+        i += 1
+    return covered_functions
+
+
+def get_slither_functions(slither, include_functions=True, include_modifiers=False, include_interfaces=False,
+                          include_variable=True):
+    # Loop for each compilation's underlying contracts
+    results = []
+
+    # Loop for each underlying contract
+    for contract in slither.contracts:
+        if not include_interfaces and contract.is_signature_only():
+            continue
+        # Loop for each function
+        functions_to_search = []
+        if include_functions:
+            functions_to_search += contract.functions
+        if include_modifiers:
+            functions_to_search += contract.modifiers
+        for function in functions_to_search:
+            if function.is_constructor or (not include_interfaces and function.is_empty is not False):
+                continue
+            results.append((contract, function))
+
+        # Loop for each state variable
+        if include_variable:
+            for variable in contract.variables:
+                if variable.visibility not in ['public', 'external']:
+                    continue
+                results.append((contract, variable))
+
+    return results
+
+
+def _get_functions_reached(compiled_functions, origin_func_desc, results):

```

Comment:
Can you describe what is the purpose of this function?

We might be able to use `function.all_high_level_calls()` and similar to avoid the recursion